README.md

```diff
@@ -40,141 +40,141 @@ npm run lint-fix # format source files
+| [#139][#139] | [SHOULD model complete business processes][#139] | - | - |
+| [#140][#140] | [SHOULD define _useful_ resources][#140] | - | - |
+| [#141][#141] | [MUST keep URLs verb-free][#141] | - | A |
+| [#142][#142] | [MUST use domain-specific resource names][#142] | - | - |
+| [#143][#143] | [MUST identify resources and sub-resources via path segments][#143] | - | - |
+| [#144][#144] | [SHOULD only use UUIDs if necessary][#144] | - | - |
+| [#145][#145] | [MAY consider using (non-)nested URLs][#145] | - | - |
+| [#146][#146] | [SHOULD limit number of resource types][#146] | - | ? |
+| [#147][#147] | [SHOULD limit number of sub-resource levels][#147] | - | B |
+| [#148][#148] | [MUST use HTTP methods correctly][#148] | - | A |
+| [#149][#149] | [MUST fulfill common method properties][#149] | - | - |
+| [#150][#150] | [MUST use standard HTTP status codes][#150] | - | - |
+| [#151][#151] | [MUST specify success and error responses][#151] | - | - |
+| [#152][#152] | [MUST use code 207 for batch or bulk requests][#152] | - | C |
+| [#153][#153] | [MUST use code 429 with headers for rate limits][#153] | - | C |
+| [#154][#154] | [MUST define collection format of header and query parameters][#154] | - | - |
+| [#155][#155] | [SHOULD reduce bandwidth needs and improve responsiveness][#155] | - | - |
+| [#156][#156] | [SHOULD use `gzip` compression][#156] | - | - |
+| [#157][#157] | [SHOULD support partial responses via filtering][#157] | - | - |
+| [#158][#158] | [SHOULD allow optional embedding of sub-resources][#158] | - | - |
+| [#159][#159] | [MUST support pagination][#159] | - | - |
+| [#160][#160] | [SHOULD prefer cursor-based pagination, avoid offset-based pagination][#160] | - | - |
+| [#161][#161] | [SHOULD use pagination links where applicable][#161] | - | - |
+| [#162][#162] | [MUST use REST maturity level 2][#162] | - | - |
+| [#163][#163] | [MAY use REST maturity level 3 - HATEOAS][#163] | - | - |
+| [#164][#164] | [MUST use common hypertext controls][#164] | - | - |
+| [#165][#165] | [SHOULD use simple hypertext controls for pagination and self-references][#165] | - | - |
+| [#166][#166] | [MUST not use link headers with JSON entities][#166] | - | - |
+| [#167][#167] | [MUST use JSON to encode structured data][#167] | - | - |
+| [#168][#168] | [MAY use non JSON media types for binary data or alternative content representations][#168] | - | - |
+| [#169][#169] | [MUST use standard date and time formats][#169] | - | A |
+| [#170][#170] | [SHOULD use standards for country, language and currency codes][#170] | - | B |
+| [#171][#171] | [MUST define format for number and integer types][#171] | - | - |
+| [#172][#172] | [SHOULD prefer standard media type name `application/json`][#172] | - | B |
+| [#173][#173] | [MUST use the common money object][#173] | - | - |
+| [#174][#174] | [MUST use common field names and semantics][#174] | - | - |
+| [#176][#176] | [MUST support problem JSON][#176] | - | - |
```

Comment:
```suggestion
| [#176][#176] | [MUST support problem JSON][#176] | - | A |
```
Otherwise we'll quickly have n different variants of how we handle api error details.

Reply:
May we should change this rule to should provide an error message in json in case of a problem

Reply:
Why "should"? In the end it's just a mapping from Exception to JSON that you need anyway. And at least for unknown Exceptions there can be a fallback to 500 with JSON content "The service made a boo boo" -> rule satisfied.

Reply:
Yes you are right, we need a must for this, not a should. But what do you think of the precision to
"MUST support a json error message for problems"

Reply:
You mean the content? If we follow https://opensource.zalando.com/problem/schema.yaml even an empty json object seems to be valid (non of the properties is marked as `required`). But it should be easy for everyone to map from an Exception to at least
- status = http code
- title = http code name (e.g. "Bad Request")
- detail = Exception text

This is about the same granularity we currently have with our web services. At least the consumer can look for those common properties and log something or construct a new exception on the client side. Of course, every service provider can provide more detailed errors - and document them additionally.

We can also derive a second ruleset from the current zalando flavoured one and add some baloise specifics if we want to be more specific about the error format.